src/lms-harmonizer/tests_integration_sql/procs/mssql/harmonize_assignment_submissions/test_lmsassignment_submissions.py

```diff
@@ -0,0 +1,251 @@
+)
+USER_TEST_EMAIL = "test@email.email"
+
+
+def describe_when_lms_and_ods_tables_are_both_empty():
+    def it_should_run_successfully(test_db_config: ServerConfig):
+        # act
+        run_harmonizer(test_db_config)
+        # assert - no errors
+
+
+def describe_when_there_are_assignment_submissions_to_insert():
+    SIS_SECTION_ID = "sis_section_id"
+    ASSIGNMENT_SOURCE_SYSTEM_IDENTIFIER = "assignment_identifier"
+    ASSIGNMENT_CATEGORY = "test_category"
+    ASSIGNMENT_SUBMISSION_STATUS = "test_submission_status"
+    USER_SIS_ID = "test_sis_id"
+    SUBMISSION_TEST_IDENTIFIER = "submission_test_identifier"
+
+    def it_should_insert_the_submissions_successfully(test_db_config: ServerConfig):
+        # arrange
+        with MSSqlConnection(test_db_config).pyodbc_conn() as connection:
+
+            insert_descriptor(connection, DESCRIPTOR_NAMESPACE, ASSIGNMENT_CATEGORY)
+            insert_lmsx_assignmentcategory_descriptor(connection, 1)
+
+            insert_descriptor(connection, DESCRIPTOR_NAMESPACE, SOURCE_SYSTEM)
+            insert_lmsx_sourcesystem_descriptor(connection, 2)
+
+            insert_descriptor(
+                connection,
+                SUBMISSION_STATUS_DESCRIPTOR_NAMESPACE,
+                ASSIGNMENT_SUBMISSION_STATUS,
+            )
+            insert_lmsx_assignmentsubmissionstatus_descriptor(connection, 3)
+
+            insert_lms_section(connection, SIS_SECTION_ID, SOURCE_SYSTEM)
+            insert_edfi_section(connection, SIS_SECTION_ID)
+            connection.execute(
+                """UPDATE LMS.LMSSECTION SET
+                    EdFiSectionId = (SELECT TOP 1 ID FROM EDFI.SECTION)"""
+            )
+
+            insert_lms_assignment(
+                connection,
+                ASSIGNMENT_SOURCE_SYSTEM_IDENTIFIER,
+                SOURCE_SYSTEM,
+                1,
+                ASSIGNMENT_CATEGORY,
+            )
+
+            insert_lms_user(connection, USER_SIS_ID, USER_TEST_EMAIL, SOURCE_SYSTEM)
+            insert_edfi_student(connection, USER_SIS_ID)
+            connection.execute(
+                """UPDATE LMS.LMSUSER SET
+                    EdFiStudentId = (SELECT TOP 1 ID FROM EDFI.Student)"""
+            )
+
+            insert_lms_assignment_submissions(
+                connection,
+                SUBMISSION_TEST_IDENTIFIER,
+                1,
+                1,
+                ASSIGNMENT_SUBMISSION_STATUS,
+                SOURCE_SYSTEM,
+                False,
+            )
+
+        # act
+        run_harmonizer(test_db_config)
+
+        # assert
+        with MSSqlConnection(test_db_config).pyodbc_conn() as connection:
+            LMSAssignmentSubmission = query(
+                connection, "SELECT * from [lmsx].[AssignmentSubmission]"
+            )
+
+            assert len(LMSAssignmentSubmission) == 1
+            assert (
+                int(LMSAssignmentSubmission[0]["AssignmentSubmissionIdentifier"]) == 1
```

Comment:
Modify mssql_loader.insert_lms_assignment_submissions() to also take an AssignmentSubmissionIdentifier (see mssql_loader.insert_edfi_student_with_usi on allowing identity inserts). Then you can use a constant ASSIGNMENT_SUBMISSION_IDENTIFIER to test against (value like 99 or something) that can't be an accidental match like 1 could be.

Reply:
It is a little bit difficult, it is currently using the value that is assigned in the lms schema by the identity column, so it will fail if I change it

Reply:
Did you try "SET IDENTITY_INSERT lms.AssignmentSubmission ON;" like in mssql_loader.insert_edfi_student_with_usi?

Reply:
It worked fine 🤓